src/integrationTest/java/org/opensearch/security/rest/WhoAmITests.java

```diff
@@ -169,4 +182,64 @@ public void testWhoAmIPost() {
         }
 
     }
+
+    @Test
+    public void testAuditLogSimilarityWithTransportLayer() {
+        try (TestRestClient client = cluster.getRestClient(AUDIT_LOG_VERIFIER)) {
+            assertThat(client.get(WHOAMI_PROTECTED_ENDPOINT).getStatusCode(), equalTo(HttpStatus.SC_OK));
+
+            auditLogsRule.assertExactly(
+                1,
+                auditPredicate(GRANTED_PRIVILEGES).withLayer(AuditLog.Origin.REST)
+                    .withRestMethod(RestRequest.Method.GET)
+                    .withRequestPath("/" + WHOAMI_PROTECTED_ENDPOINT)
+                    .withEffectiveUser(AUDIT_LOG_VERIFIER)
+            );
+
+            assertThat(client.get("_cat/indices").getStatusCode(), equalTo(HttpStatus.SC_OK));
+
+            auditLogsRule.assertExactly(2, grantedPrivilege(AUDIT_LOG_VERIFIER, "GetSettingsRequest"));
+
+            List<AuditMessage> grantedPrivilegesMessages = auditLogsRule.getCurrentTestAuditMessages()
+                .stream()
+                .filter(msg -> msg.getCategory().equals(GRANTED_PRIVILEGES))
+                .collect(Collectors.toList());
+            verifyAuditLogSimilarity(grantedPrivilegesMessages);
+        }
+    }
+
+    private void verifyAuditLogSimilarity(List<AuditMessage> currentTestAuditMessages) {
+        List<AuditMessage> restSet = new ArrayList<>();
+        List<AuditMessage> transportSet = new ArrayList<>();
+
+        // It is okay to loop through all even though we end up using only 2, as the total number of messages should be around 8
+        for (AuditMessage auditMessage : currentTestAuditMessages) {
+            if ("REST".equals(auditMessage.getAsMap().get(AuditMessage.REQUEST_LAYER).toString())) {
+                restSet.add(auditMessage);
+            } else if ("TRANSPORT".equals(auditMessage.getAsMap().get(AuditMessage.REQUEST_LAYER).toString())) {
+                transportSet.add(auditMessage);
+            }
+        }
+        // We pass 1 message from each layer to check for similarity
+        checkForStructuralSimilarity(restSet.get(0), transportSet.get(0));
+    }
+
+    private void checkForStructuralSimilarity(AuditMessage restAuditMessage, AuditMessage transportAuditMessage) {
```

Comment:
Can you provide a sample of each type of audit log message? restAuditMessage and transportAuditMessage?